torch_audiomentations/augmentations/impulse_response.py

```diff
@@ -11,21 +11,36 @@ class ApplyImpulseResponse(BasicTransform):
     Convolves an input signal with an impulse response.
     """
 
-    def __init__(self, ir_path, convolve_mode="full", p=0.5):
+    def __init__(self, ir_path, device=torch.device("cpu"), convolve_mode="full", p=0.5):
         super(ApplyImpulseResponse, self).__init__(p)
         self.ir_path = find_audio_files(ir_path)
+
+        if len(self.ir_path) == 0:
+            raise EmptyPathException("There are no supported audio files found.")
+
         self.convolve_mode = convolve_mode
+        self.device = device
 
     def randomize_parameters(self, samples, sample_rate):
         super(ApplyImpulseResponse, self).randomize_parameters(samples, sample_rate)
-        if self.parameters["should_apply"]:
-            self.parameters["ir_file_path"] = random.choice(self.ir_path)
-
-    def forward(self, samples, sample_rate):
-        super(ApplyImpulseResponse, self).forward(samples, sample_rate)
-
-        if self.parameters["should_apply"] and len(samples) > 0:
-            ir = torch.from_numpy(load_audio(self.parameters["ir_file_path"], sample_rate))
-            return convolve(samples, ir, mode=self.convolve_mode)
+        if self.parameters["should_apply"] and samples.size(0) > 0:
+            ir_paths = random.choices(self.ir_path, k=samples.size(0))
+            ir_sounds = []
+            max_ir_sound_length = 0
+            for ir_path in ir_paths:
+                ir_samples = load_audio(ir_path, sample_rate)
+                num_samples = len(ir_samples)
+                if num_samples > max_ir_sound_length:
+                    max_ir_sound_length = num_samples
```

Comment:
A bit clearer I guess. 
```suggestion
                max_ir_sound_length = max(len(ir_samples), max_ir_sound_length)
```